sailon/random_novelty_detector.py

```diff
@@ -35,21 +27,10 @@ def get_config(self):
         """
         return {}
 
-    def execute(self, step_descriptor: str, *args, **kwargs):
-        """
-        Execute method used by the protocol to run different steps associated
-        with the algorithm.
-
-        Args:
-            step_descriptor (str): Name of the step.
-        """
-        logging.info(f"Executing {step_descriptor}")
-        return self.step_dict[step_descriptor](*args, **kwargs)
-
-    def _initialize(self, config: dict):
+    def initialize(self, config: dict):
         self.config = config
 
-    def _feature_extraction(self, fpaths: List[str]):
+    def feature_extraction(self, fpaths: List[str]):
```

Comment:
The parameters don't match the calls in the protocol. Same comment for the other function calls too.

Reply:
Corrected the method signatures in the detector class definition and method calls in the protocol.